core/utils/dom.js

```diff
@@ -72,16 +72,132 @@ Blockly.utils.dom.cacheReference_ = 0;
  */
 Blockly.utils.dom.canvasContext_ = null;
 
+/**
+ * A name with the type of the SVG element stored in the generic.
+ * @param {string} tagName The SVG element tag name.
+ * @constructor
+ * @template T
+ */
+Blockly.utils.dom.SVGElementType = function(tagName) {
+  /**
+   * @type {string}
+   * @private
+   */
+  this.tagName_ = tagName;
+};
+
+/**
+ * Returns the SVG element tag name.
+ * @return {string} The name.
+ * @override
+ */
+Blockly.utils.dom.SVGElementType.prototype.toString = function() {
+  return this.tagName_;
+};
+
+/** @type {!Blockly.utils.dom.SVGElementType<!SVGAnimateElement>} */
+Blockly.utils.dom.SVGElementType.SVGAnimateElement =
```

Comment:
Since  'SVGElement' is already in the name, do you think it might be more clear to have final part of the path be just the element name? e.g. `Blockly.utils.dom.SvgElementType.Animate`?
I find 'SVGSVGElement` and `SVGGElement` both really hard to read and this would solve it for both of them and make all of it just a bit less verbose.

Reply:
Agreed it's long. My concern is it's not easily derivable at the call site what type of element you are getting back. This way, you know that calling `Blockly.createSvgElement` with name `Blockly.utils.dom.SVGElementType.SVGAnimateElement` is going to yield an `SVGAnimateElement` type.

I would be open to changing the `Blockly.utils.dom.SVGElementType` to something else if you think of an alternative there.

Reply:
Oh. On second thought, this doesn't make sense at all.

If you were used to doing: 
```Blockly.utils.dom.createSvgElement('g'`` Now you need to know that g is in fact an SVGGElement. That's bizarre, so good point. Changing them all to use the name.